### PR TITLE
Adds versionbit rejection code

### PR DESCRIPTION
**Pull request by red010b37**
This PR introduces the concept of bit version rejection. 

Please not this is a resubission of https://github.com/aguycalled/navcoin-core/pull/14 with the changes and code spit out as requested!

It is designed to make it easier for the network to reject individual soft forks when they are bundled together in 1 release. As soft forks come within the software signaling by default only the converse was needed for people to reject soft forks they did not agree with.

A new config concept has been added called `rejectversionbit`  users can signal all the soft forks they reject by adding the following to the config

```
rejectversionbit=15
rejectversionbit=16
rejectversionbit=17
```

I built this as a new function in versionbits.cpp as that seemed the most logical place instead of mashing it into the ComputeBlockVersion, I think this will allows some expansion later down the line

### CODE REVIEW

File: src/main.cpp
Comment (proletesseract) [inline src/main.cpp:2473]: I'll just re-ask this question here since I'm not sure it was in the right line on the last PR. This loop is looping through all the version bits defined in consensus/params.h and then the IsVersionBitRejected function loops through each of the values in the config to check if something is rejected. Are there any of these which we don't want people to be able to disable;

https://github.com/aguycalled/navcoin-core/blob/5a0aa5493388076e50b5c671c3f9dd80618e3aef/src/consensus/params.h#L16

I assume if someone turns off the segwit bit for example, it will have no effect. But it would be good to double check with you guys.

Comment (red010b37) [reply src/main.cpp:2473]: As far as I understand, this won't affect any of the bit versions that have already been accepted and activated,

@aguycalled ^

Comment (aguycalled) [reply src/main.cpp:2473]: it's more of a formal aspect. you can change the logic to something like
```
if (state == THRESHOLD_LOCKED_IN || state == THRESHOLD_ACTIVE || (state == THRESHOLD_STARTED
                 && !IsVersionBitRejected(params, (Consensus::DeploymentPos)i)))) {
```
so IsVersionBitRejected() only applies on the STARTED state.
with the logic proposed by @red010b37 the bits would also be affected in the other states, (without any practical consequence as the necessary bit masks are set after this loop).
for sanity i consider IsVersionBitRejected should only affect when on the STARTED state

Comment (red010b37) [reply src/main.cpp:2473]: @aguycalled that makes sense, I will make the change suggested

Comment (red010b37) [reply src/main.cpp:2473]: @craigmacgregor @aguycalled I have updated the code to reflect the conversation above

File: src/versionbits.cpp
Comment (aguycalled) [inline src/versionbits.cpp]: remember cleaning this log before merging

**Comment by aguycalled** [comment]
i've added a test and it passes. 
```
reject-version-bit.py:
Initializing test directory /var/folders/qh/7_l8lmqx2sb5ncxgtzzyb51r0000gn/T/testmko6ph41/0
Mining blocks...
Stopping nodes
Cleaning up
Tests successful

 
Pass: True, Duration: 25 s

TEST                  | PASSED | DURATION

reject-version-bit.py | True   | 25 s

ALL                   | True   | 25 s (accumulated)

Runtime: 25 s

```
i'll merge if you guys are happy with the test i wrote

**Review by aguycalled — APPROVED** [review]

**Comment by red010b37** [comment]
@aguycalled added the help text, test are passing ;)

**Pull request merged**
